### PR TITLE
feat(web-components): add large variant to status tag

### PR DESCRIPTION
**Pull request by GCHQ-Developer-773**
added large variant to status tag and added large to storybook, and included large and large
outlined in css for status    tags

## Summary of the changes
Tell us, in a few words, what the changes are.

## Related issue
#239

### General 

- [ ] Changes to docs package checked and committed.
- [/ ] All acceptance criteria reviewed and met. 

### Testing

- [ ] Relevant unit tests and visual regression tests added. 
- [ ] Visual testing against Figma component specification completed. 

### Accessibility 

- [ ] Accessibility Insights FastPass performed.
- [ ] A11y unit test added and yields no issues.
- [ ] A11y plug-in on Storybook yields no issues. 
- [ ] Manual screen reader testing performed using NVDA and VoiceOver. 
- [ ] Manual keyboard testing for keyboard controls and logical focus order. 
- [ ] Correct roles used and ARIA attributes used correctly where required. 
- [ ] Logical heading structure is maintained, and the HTML elements used for headings can be changed to fit within the wider page structure. 

### Resize/zoom behaviour 

- [/ ] Page can be zoomed to 400% with no loss of content. 
- [ /] Screen magnifier used with no issues. 
- [/ ] Text resized to 200% with no loss of content.
- [ ] Text spacing increased as per the [WCAG 1.4.12 success criterion](https://www.w3.org/TR/WCAG21/#text-spacing) with no loss of content.

### System modes

- [/ ] Browser setting 'prefers reduced motion' tested. No animations or motion visible whilst this setting is on. 
- [/ ] Windows High Contrast mode tested with no loss of content.
- [/ ] System light and dark mode tested with no loss of content.
- [/ ] Browser support tested (Chrome, Safari, Firefox and Edge). 

### Testing content extremes

- [ ] Min/max content examples tested with no loss of content or overflow. 
- [ ] All prop combinations work without issue. 
- [ ] Tested for FOUC (Flash of Unstyled Content) in both SSR (Server-Side Rendering) and SSG (Static Site Generation) settings.
- [ ] Controlled and uncontrolled input components tested.

### CODE REVIEW

File: packages/web-components/src/components/ic-status-tag/ic-status-tag.tsx [hunk 1]
```diff
@@ -71,7 +71,9 @@ export class StatusTag {
             [`${appearance}-${status}`]: appearance !== undefined,
             [`${variant}-${status}`]: true,
             ["outlined"]: variant === "outlined",
-            ["small"]: small || size === "small",
```
Comment (GCHQ-Developer-112) [inline packages/web-components/src/components/ic-status-tag/ic-status-tag.tsx:74]: Some, if not all of this line should still be in the code. Using the deprecated prop: `small` should still have an effect on the size of the status tag for the time being. This prop will be removed in V3

File: packages/web-components/src/components/ic-status-tag/ic-status-tag.tsx [hunk 2]
```diff
@@ -71,7 +71,9 @@ export class StatusTag {
...
+            [`${size}`]: true,
+
+
```
Comment (GCHQ-Developer-112) [inline packages/web-components/src/components/ic-status-tag/ic-status-tag.tsx:76]: Please remove these extra lines

**Comment by GCHQ-Developer-112** [comment]
Sorry, just realised that this is a draft PR so you may already be working to address the comments above. Please add a story to the React storybook if you're not already working on this

File: packages/react/src/stories/ic-status-tag.stories.mdx [hunk 3]
```diff
@@ -55,6 +55,16 @@ import readme from "../../../web-components/src/components/ic-status-tag/readme.
   <Story name="Small">
     <IcStatusTag label="Small" size="small" />
     <IcStatusTag label="Small" variant="outlined" size="small" />
+    <IcStatusTag label="Small" small="true" />
+  </Story>
```
Comment (GCHQ-Developer-112) [inline packages/react/src/stories/ic-status-tag.stories.mdx:59]: Please can you combine this into one 'Sizes story' - similar to this [ic-button size story](https://mi6.github.io/ic-ui-kit/branches/develop/web-components/?path=/story/web-components-button--size). This applies to the web components storybook as well

**Review by GCHQ-Developer-112 — CHANGES_REQUESTED** [review]
Looks good overall!
Please can you add a visual regression Cypress test for the new size variant

File: packages/web-components/src/components/ic-status-tag/ic-status-tag.css [hunk 4]
```diff
@@ -15,6 +15,16 @@
   border-radius: calc(var(--ic-space-md) - var(--ic-space-xxxs));
 }
 
+.large {
+  padding: var(--ic-space-xs) var(--ic-space-sm);
+  border-radius: calc(var(--ic-space-md) + var(--ic-space-xxs));
+}
+
+.large {
```
Comment (GCHQ-Developer-112) [inline packages/web-components/src/components/ic-status-tag/ic-status-tag.css:23]: Please remove this duplicate CSS style

**Review by GCHQ-Developer-112 — CHANGES_REQUESTED** [review]
One small change needed, then all good for approval 🌟

**Comment by GCHQ-Developer-530** [comment]
Please can you update both web components and react playground stories to have large as a size option?

**Comment by GCHQ-Developer-773** [comment]
@GCHQ-Developer-530 all sizes can be both in both stories under the sizes story. Any issues just let me know :)

**Comment by GCHQ-Developer-530** [comment]
> @GCHQ-Developer-530 all sizes can be both in both stories under the sizes story. Any issues just let me know :)

It needs updating in the playground story as well so we can see the large variant. Should just be a case of adding large as an option for this line: 
```
options: [
          "default", "small"
        ],
```

**Review by GCHQ-Developer-530 — DISMISSED** [review]

**Review by GCHQ-Developer-530 — APPROVED** [review]

**Review by GCHQ-Developer-112 — APPROVED** [review]

**Pull request merged**